streaming/src/main/scala/io/gearpump/streaming/StreamApplication.scala

```diff
@@ -37,9 +39,11 @@ trait Processor[+T <: Task] extends ReferenceEqual {
 
   def description: String
 
-  def taskClass: Class[_ <: Task]
+  def taskClass: Class[_<:T]
```

Comment:
This is wrong.

1. 
For example, if we have task ABCTask, then the taskClass should return Class[ABCTask] instead of Class[? <: ABCTask]

2. The reason that we don't use Class[T] but Class[_ <: Task] is that Class is a class defined in java, which don't have the concept of covariant or contravariant. By erasuing the actual task type, it gives us some flexibility.